KCAL-77 When an assigned BLF used for parking is pressed with out an …

enoskov-at-ooma opened the pull request:
KCAL-77 When an assigned BLF used for parking is pressed with out an active call it places the Device in the Parking spot

Fixing the following scenario:
1. User A dials *31 (auto park to slot 1). A call is placed to slot 1 and it waits for 30 seconds for a redirected call as a attendant transfer sequence.
2. Before 30 seconds expire User B dials *31 as well and it finds an occupied slot. But it is in attendant state. So it retries 3 times with 1 sec interval.
3. After 3rd retry it ignores attendant state and connects. As a result user A and B are connected.
4. If, while in active call between A and B, User A dials *31 again in second line, the call is dropped and park slot is occupied (limbo state). 

If it is still in attendant state after 3 retries, return error (play occupied and exit).

Test plan:
 
Test Case 1
----------- 
Actions: 
1. Make a call on user A to *31 (auto). 
2. Before 30 sec expiration period, user B dials *31
3. User B dials *31 again.
4. User B dials *51 (retrieve)
5. 30 seconds expired for user A.
Expected:
1. A call is placed to parking spot 1 ivr is played to user A
2. Slot is occupied is played to user B.
3. Slot is occupied is played to user B again.
4. This parking extension is empty is played to user B.
5. The attended call from user A is cleared automatically after 30 seconds as no replacing INVITE is received as a part of attendant transfer sequence. The spot is cleared. No entry in couchdb in limbo state.

enoskov-ooma committed:
KCAL-77 When an assigned BLF used for parking is pressed with out an active call it places the Device in the Parking spot

Fixing the following scenario:
1. User A dials *31 (auto park to slot 1). A call is placed to slot 1 and it waits for 30 seconds for a redirected call
   as a attendant transfer sequence.
2. Before 30 seconds expire User B dials *31 as well and it finds an occupied slot. But it is in attendant state.
   So it retries 3 times with 1 sec interval.
3. After 3rd retry it ignores attendant state and connects. As a result user A and B are connected.
4. If, while in active call between A and B, User A dials *31 again in second line, the call is dropped and park slot is
   occupied (limbo state).
After discussion with @luis.azedo it was decided to not even retry 3 times.
If its in attendant state return error (play occupied and exit).

enoskov-at-ooma committed:
Merge branch 'master' into KCAL-77

lazedo reviewed (changes requested):
`attended` change looks good.
please keep the code for retrying

enoskov-ooma committed:
KCAL-77 When an assigned BLF used for parking is pressed with out an active call it places the Device in the Parking spot

Fixing the following scenario:
1. User A dials *31 (auto park to slot 1). A call is placed to slot 1 and it waits for 30 seconds for a redirected call
 as a attendant transfer sequence.
2. Before 30 seconds expire User B dials *31 as well and it finds an occupied slot. But it is in attendant state.
 So it retries 3 times with 1 sec interval.
3. After 3rd retry it ignores attendant state and connects. As a result user A and B are connected.
4. If, while in active call between A and B, User A dials *31 again in second line, the call is dropped and park slot is
 occupied (limbo state).
If it is still in attendant state after 3 retries, return error (play occupied and exit).

enoskov-at-ooma commented:
> `attended` change looks good. please keep the code for retrying

done
